src/core_plugins/kibana/public/management/sections/indices/create_index_pattern/create_index_pattern.js

```diff
@@ -52,97 +187,108 @@ uiModules.get('apps/management')
-        if (id) {
-          refreshKibanaIndex().then(function () {
-            if (!config.get('defaultIndex')) {
-              config.set('defaultIndex', indexPattern.id);
-            }
-            indexPatterns.cache.clear(indexPattern.id);
-            kbnUrl.change('/management/kibana/indices/' + indexPattern.id);
-          });
+      refreshKibanaIndex().then(function () {
+        if (!config.get('defaultIndex')) {
+          config.set('defaultIndex', id);
         }
-      });
 
-      // refreshFields calls save() after a successfull fetch, no need to save again
-      // .then(function () { indexPattern.save(); })
-    })
-    .catch(function (err) {
+        indexPatterns.cache.clear(id);
+        kbnUrl.change(`/management/kibana/indices/${id}`);
+      });
+    }).catch(err => {
       if (err instanceof IndexPatternMissingIndices) {
-        notify.error($translate.instant('KIBANA-NO_INDICES_MATCHING_PATTERN'));
+        return notify.error($translate.instant('KIBANA-NO_INDICES_MATCHING_PATTERN'));
       }
-      else notify.fatal(err);
+
+      notify.fatal(err);
     });
   };
 
-
   $scope.$watchMulti([
-    'index.isTimeBased',
-    'index.nameIsPattern',
-    'index.nameInterval.name'
+    'newIndexPattern.isTimeBased',
+    'newIndexPattern.nameIsPattern',
+    'newIndexPattern.nameInterval.name'
   ], function (newVal, oldVal) {
     const isTimeBased = newVal[0];
     const nameIsPattern = newVal[1];
-    const newDefault = getPatternDefault(newVal[2]);
-    const oldDefault = getPatternDefault(oldVal[2]);
+    const newDefault = getDefaultPatternForInterval(newVal[2]);
+    const oldDefault = getDefaultPatternForInterval(oldVal[2]);
 
-    if (index.name === oldDefault) {
-      index.name = newDefault;
+    if ($scope.newIndexPattern.name === oldDefault) {
+      $scope.newIndexPattern.name = newDefault;
     }
 
     if (!isTimeBased) {
-      index.nameIsPattern = false;
+      $scope.newIndexPattern.nameIsPattern = false;
     }
 
     if (!nameIsPattern) {
-      delete index.nameInterval;
-      delete index.timeField;
+      delete $scope.newIndexPattern.nameInterval;
+      delete $scope.newIndexPattern.timeField;
     } else {
-      index.nameInterval = index.nameInterval || intervals.byName.days;
-      index.name = index.name || getPatternDefault(index.nameInterval);
+      $scope.newIndexPattern.nameInterval = $scope.newIndexPattern.nameInterval || intervals.byName.days;
+      $scope.newIndexPattern.name = $scope.newIndexPattern.name || getDefaultPatternForInterval($scope.newIndexPattern.nameInterval);
     }
   });
 
   $scope.moreSamples = function (andUpdate) {
-    index.sampleCount += 5;
+    $scope.ctrl.sampleCount += 5;
     if (andUpdate) updateSamples();
   };
 
   $scope.$watchMulti([
-    'index.name',
-    'index.nameInterval'
+    'newIndexPattern.name',
+    'newIndexPattern.nameInterval'
   ], function (newVal, oldVal) {
+    function promiseMatch(lastPromise, cb) {
```

Comment:
It's not well named, and it relies on external values, neither are your fault, but I can understand the confusion. Still, now this function is being defined every time the `watchMulti` is triggered, which may not be terribly expensive, but it isn't free either. 

If you were to, for example, define this function in a for or while loop, the linter would yell at you. This code isn't that different, the function is still redefined and garbage collected repeatedly. Not cause for changing it, perhaps, just something to note. That's likely the reason it was defined externally originally.